packages/language-picker/src/language-picker.tsx

```diff
@@ -0,0 +1,150 @@
+
+/**
+ * Internal dependencies
+ */
+import { Language, LanguageGroup } from './Language';
+import { getSearchedLanguages, LocalizedLanguageNames } from './search';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+type Props = {
+	onSelectLanguage: ( language: Language ) => void;
+	languages: Language[];
+	languageGroups: LanguageGroup[];
+	defaultLananguageGroupId: string;
+	selectedLanguage?: Language;
+	search?: string;
+	localizedLanguageNames?: LocalizedLanguageNames;
+};
+
+const findBestDefaultLanguageGroupId = (
+	selectedLanguage: Language | undefined,
+	languageGroups: LanguageGroup[],
+	defaultLananguageGroupId: string
+): string => {
+	if ( ! selectedLanguage ) {
+		return defaultLananguageGroupId;
+	}
+
+	if ( selectedLanguage.popular ) {
+		return 'popular';
+	}
+
+	return (
+		languageGroups.find( ( lg ) => {
+			const sharedTerritories = intersection( lg.subTerritories, selectedLanguage.territories );
+
+			if ( sharedTerritories.length > 0 ) {
+				return lg;
+			}
+
+			return false;
+		} )?.id ?? defaultLananguageGroupId
+	);
+};
+
+const LanguagePicker = ( {
+	onSelectLanguage,
+	languages,
+	languageGroups,
+	defaultLananguageGroupId,
+	selectedLanguage,
+	search,
+	localizedLanguageNames,
+}: Props ) => {
+	const { __ } = useI18n();
+	const [ filter, setFilter ] = useState(
+		findBestDefaultLanguageGroupId( selectedLanguage, languageGroups, defaultLananguageGroupId )
+	);
+
+	const getFilteredLanguages = () => {
+		switch ( filter ) {
+			case 'popular':
+				return languages
+					.filter( ( language ) => language.popular )
+					.sort( ( a, b ) => ( a.popular as number ) - ( b.popular as number ) );
+			default: {
+				const languageGroup = languageGroups.find( ( l ) => l.id === filter );
+				const subTerritories = languageGroup ? languageGroup.subTerritories : [];
+				return languages
+					.filter( ( language ) =>
+						language.territories.some( ( t ) => subTerritories?.includes( t ) )
+					)
+					.sort( ( a, b ) => a.name.localeCompare( b.name ) );
+			}
+		}
+	};
+
+	const renderCategoryButtons = () => {
+		return languageGroups.map( ( languageGroup ) => {
+			const isSelected = filter === languageGroup.id;
+
+			const onClick = () => {
+				setFilter( languageGroup.id );
+			};
+
+			return (
+				<div key={ languageGroup.id }>
+					<Button onClick={ onClick } className="language-picker__language-group">
+						<span className={ isSelected ? 'is-selected' : '' }>{ languageGroup.name( __ ) }</span>
+					</Button>
+				</div>
+			);
+		} );
+	};
+
+	const shouldDisplayRegions = ! search;
+
+	const languagesToRender = search
+		? getSearchedLanguages( languages, search, localizedLanguageNames )
+		: getFilteredLanguages();
+
+	return (
+		<>
+			<div className="language-picker__labels">
+				{ shouldDisplayRegions ? (
+					<>
+						<div className="language-picker__regions-label">{ __( 'regions' ) }</div>
+						<div className="language-picker__languages-label">{ __( 'language' ) }</div>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Language', { context: 'logged-out-homepage'} )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/use-lp-in-calypso)